src/wireviz/Harness.py

```diff
@@ -330,90 +351,118 @@ def output(self, filename: (str, Path), view: bool = False, cleanup: bool = True
-                                     'pn': index_if_list(bundle.pn, index)})
-        # join similar wires from all the bundles to a single BOM item
-        wire_group = lambda w: (w.get('type', None), w['gauge'], w['gauge_unit'], w['color'], w['manufacturer'], w['mpn'], w['pn'])
-        for group in Counter([wire_group(v) for v in wirelist]):
-            items = [v for v in wirelist if wire_group(v) == group]
+        for cable in self.cables.values():
+            if not cable.ignore_in_bom:
+                # create name beilds used by both cables and bundles
+                cable_type = f', {remove_line_breaks(cable.type)}' if cable.type else ''
+                gauge_name = f' x {cable.gauge} {cable.gauge_unit}' if cable.gauge else ' wires'
+                if cable.category != 'bundle':
+                    # process cable as a single entity
+                    shield_name = ' shielded' if cable.shield else ''
+                    name = f'Cable{cable_type}, {cable.wirecount}{gauge_name}{shield_name}'
+                    item = {'item': name, 'qty': cable.length, 'unit': 'm', 'designators': cable.name,
+                            'manufacturer': remove_line_breaks(cable.manufacturer), 'mpn': remove_line_breaks(cable.mpn), 'pn': cable.pn}
+                    bom_items.append(item)
+                else:
+                    # add each wire from the bundle to the bom
+                    for index, color in enumerate(cable.colors, 0):
+                        wire_color = f', {color}' if color else ''
+                        name = f'Wire{cable_type}{gauge_name}{wire_color}'
+                        item = {'item': name, 'qty': cable.length, 'unit': 'm', 'designators': cable.name,
+                                'manufacturer': remove_line_breaks(index_if_list(cable.manufacturer, index)),
+                                'mpn': remove_line_breaks(index_if_list(cable.mpn, index)), 'pn': index_if_list(cable.pn, index)}
+                        bom_items.append(item)
+
+            for part in cable.additional_components:
+                qty = part.qty * cable.get_qty_multiplier(part.qty_multiplier)
+                bom_items.append(
+                    {
+                        'item': part.description,
+                        'qty': qty,
+                        'unit': part.unit,
+                        'manufacturer': part.manufacturer,
+                        'mpn': part.mpn,
+                        'pn': part.pn,
+                        'designators': cable.name
+                    }
+                )
+
+        for item in self.additional_bom_items:
+            item = {'item': item.get('description', ''), 'qty': item.get('qty'), 'unit': item.get('unit'), 'designators': item.get('designators'),
+                    'manufacturer': item.get('manufacturer'), 'mpn': item.get('mpn'), 'pn': item.get('pn')}
+            bom_items.append(item)
+
```

Comment:
I had missed that the remove_line_breaks function would pass cleanly over non string items. so that would have worked fine.

I have done some tests on fields with newlines. Adding newlines (or any other whitespace changes) in fields will cause bom lines to be split into multiple rows. The catch with multiline fields is that by default the | operator in yaml appends a newline on the end generating additional spaces in the descriptions when run through remove_line_breaks() and causing them not to match due to that. 

There are a few options for how to deal with this.
1. Leave it as is with items being spit /defer this to a new issue.
2. Remove whitespace during deduplication matching, this could cause the unwanted whitespace to appear in the bom descriptions as we would need to pick one to use.
3. move the linebreak code back to the individual fields

I think 3 is probably the best soloution, will implement that tomorow unless there is a preference for one of the options.